eclair-core/src/main/scala/fr/acinq/eclair/db/sqlite/SqlitePaymentsDb.scala

```diff
@@ -150,30 +158,31 @@ class SqlitePaymentsDb(sqlite: Connection) extends PaymentsDb with Logging {
       PublicKey(rs.getByteVector("target_node_id")),
       rs.getLong("created_at"),
       rs.getStringNullable("payment_request").map(PaymentRequest.read),
-      OutgoingPaymentStatus.Pending
+      status_opt.getOrElse(OutgoingPaymentStatus.Pending)
     )
-    // If we have a pre-image, the payment succeeded.
-    rs.getByteVector32Nullable("payment_preimage") match {
-      case Some(paymentPreimage) => result.copy(status = OutgoingPaymentStatus.Succeeded(
-        paymentPreimage,
-        MilliSatoshi(rs.getLong("fees_msat")),
-        rs.getBitVectorOpt("payment_route").map(b => paymentRouteCodec.decode(b) match {
+  }
+
+  private def buildOutgoingPaymentStatus(preimage_opt: Option[ByteVector32], fees_opt: Option[MilliSatoshi], paymentRoute_opt: Option[BitVector], completedAt_opt: Option[Long], failures: Option[BitVector]): Option[OutgoingPaymentStatus] = {
+    preimage_opt match {
+      // If we have a pre-image, the payment succeeded.
+      case Some(preimage) => Some(OutgoingPaymentStatus.Succeeded(
+        preimage, fees_opt.getOrElse(MilliSatoshi(0)), paymentRoute_opt.map(b => paymentRouteCodec.decode(b) match {
           case Attempt.Successful(route) => route.value
           case Attempt.Failure(_) => Nil
         }).getOrElse(Nil),
-        rs.getLong("completed_at")
+        completedAt_opt.getOrElse(0)
       ))
-      case None => getNullableLong(rs, "completed_at") match {
+      case None => completedAt_opt match {
         // Otherwise if the payment was marked completed, it's a failure.
-        case Some(completedAt) => result.copy(status = OutgoingPaymentStatus.Failed(
-          rs.getBitVectorOpt("failures").map(b => paymentFailuresCodec.decode(b) match {
-            case Attempt.Successful(failures) => failures.value
+        case Some(completedAt) => Some(OutgoingPaymentStatus.Failed(
+          failures.map(b => paymentFailuresCodec.decode(b) match {
+            case Attempt.Successful(f) => f.value
             case Attempt.Failure(_) => Nil
           }).getOrElse(Nil),
           completedAt
         ))
         // Else it's still pending.
```

Comment:
nit: that comment isn't up-to-date.
You should either remove that comment or change this method to always return an `OutgoingPaymentStatus` and use `Some(OutgoingPaymentStatus.Pending)` here instead of `None`.

Reply:
I think the better choice is to have that method always return an `OutgoingPaymentStatus` instead of an `Option` (otherwise you're duplicating the `getOrElse(OutgoingPaymentStatus.Pending)`